src/NServiceBus.Core/Recoverability/FirstLevelRetries/FirstLevelRetriesBehavior.cs

```diff
@@ -28,28 +29,26 @@ public FirstLevelRetriesBehavior(FlrStatusStorage storage, FirstLevelRetryPolicy
                 var messageId = context.Message.MessageId;
                 var pipelineUniqueMessageId = messageId;
 
-                var numberOfFailures = storage.GetFailuresForMessage(pipelineUniqueMessageId);
+                storage.MarkForFirstLevelRetry(pipelineUniqueMessageId, ExceptionDispatchInfo.Capture(ex));
```

Comment:
at first glance the only usage in SLR is this:
 `message.Headers[Headers.FLRetries] = (failureInfo.FailedProcessingAttempts - 1).ToString();`

When FLR doesn't handle the message, it also shoudln't mark this as a failed attempt since that seems to be outside of FLRs responsibility. FLR can be disabled, in that case the counter would have a different behavior than with enabled FLR, right?